cli/command/trust/signer_remove.go

```diff
@@ -0,0 +1,135 @@
+package trust
+
+import (
+	"fmt"
+
+	"os"
+
+	"github.com/docker/cli/cli"
+	"github.com/docker/cli/cli/command"
+	"github.com/docker/cli/cli/trust"
+	"github.com/docker/notary/client"
+	"github.com/docker/notary/tuf/data"
+	"github.com/spf13/cobra"
+)
+
+type signerRemoveOptions struct {
+	forceYes bool
+	images   []string
+	signer   string
+}
+
+func newSignerRemoveCommand(dockerCli command.Cli) *cobra.Command {
+	options := signerRemoveOptions{}
+	cmd := &cobra.Command{
+		Use:   "signer-remove NAME IMAGE [IMAGE...]",
+		Short: "Remove a signer",
+		Args:  cli.RequiresMinArgs(2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			options.signer = args[0]
+			options.images = args[1:]
+			return removeSigner(dockerCli, &options)
+		},
+	}
+	flags := cmd.Flags()
+	flags.BoolVarP(&options.forceYes, "yes", "y", false, "Answer yes to removing most recent signer (no confirmation)")
+	return cmd
+}
+
+func removeSigner(cli command.Cli, options *signerRemoveOptions) error {
+	if len(options.images) < 1 {
+		return fmt.Errorf("at least one image must be provided")
+	}
+
+	for _, image := range options.images {
+		if err := removeSingleSigner(cli, image, options.signer, options.forceYes); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func isLastSigner(role data.Role, notaryRepo *client.NotaryRepository) (bool, error) {
+	allRoles, err := notaryRepo.ListRoles()
+	if err != nil {
+		return true, err
+	}
+	var lastSigner client.RoleWithSignatures
+	for i := range allRoles {
+		if allRoles[i].Name == "targets/releases" {
+			lastSigner = allRoles[i]
+		}
+	}
+	if len(lastSigner.Signatures) == 0 {
+		fmt.Printf("All tags are currently unpullable, use docker trust sign to fix")
+		return false, nil
+	}
+	nonRoleKey := false
+	for _, signature := range lastSigner.Signatures {
+		for _, key := range role.KeyIDs {
+			if signature.KeyID == key {
+				nonRoleKey = true
+				break
+			}
+		}
+		if nonRoleKey {
+			break
+		}
+	}
+	return nonRoleKey, nil
+}
+
+func removeSingleSigner(cli command.Cli, image, signerName string, forceYes bool) error {
+	_, ref, repoInfo, authConfig, err := getImageReferencesAndAuth(cli, image)
+	if err != nil {
+		return err
+	}
+	notaryRepo, err := trust.GetNotaryRepository(cli, repoInfo, *authConfig, "push", "pull")
+	if err != nil {
+		return trust.NotaryError(ref.Name(), err)
+	}
+	delegation := "targets/" + signerName
+	var role data.Role
+	if delegation == releases {
+		//return some error
```

Comment:
we should mirror the error on `signer-add`.  Best way to do this is probably to create a new error type and share them between the two commands